### DISCUSSION TITLE
How to save output as mp4 and understanding ffmpeg options

### DISCUSSION THREAD

**Discussion by Infinitay** [streamlink/streamlink#6026] [Q&A] [answered]
### Description

I've been using yt-dlp for Twitch for a while now but I finally want to give streamlink a try. I've set up most of my config, but I'm trying to figure out how to save the output as an `mp4` as yt-dlp does it automatically. I was looking at the docs but I'm having troubling understanding which option I should use but more importantly if the command runs at the end. 

- [--ffmpeg-fout OUTFORMAT](https://streamlink.github.io/cli.html#cmdoption-ffmpeg-fout)
	- When muxing streams, set the output format to OUTFORMAT. Default is: "matroska". Example: --ffmpeg-fout "mpegts"
- [--ffmpeg-video-transcode CODEC](https://streamlink.github.io/cli.html#cmdoption-ffmpeg-video-transcode)
	- When muxing streams, transcode the video to CODEC. Default is: "copy". Example: --ffmpeg-video-transcode "h264"

I'm thinking that I only need to use `--ffmpeg-fout mpeg-4`  and leaving the transcode option as default `--ffmpeg-video-transcode copy` because I don't want to transcode it, and I want the original file just as a mp4 instead. Am I misunderstanding the transcoding and the commands?

Thanks

**Comment by bastimeyer** [comment, marked as answer]
Streamlink only muxes the stream output if it has to, e.g. when muxing multiple sources. This is not the case here, hence the emphasis on "when" in the docs. The options have no effect here. Since recording is not the main purpose of Streamlink, there's no focus on any muxing / output format or re-encoding options.

If you look on the issue tracker and discussion forum, this has been answered dozens of times.

```sh
# on the fly
streamlink --stdout URL best | ffmpeg -i pipe:0 -c copy out.mp4

# post-download
streamlink -o out.ts URL best && ffmpeg -i out.ts -c copy out.mp4 && rm out.ts
```

**Reply by Infinitay** [reply]
Apologies for asking a common question and thanks for the answer
